### PR TITLE
avoid mpileup writing to inputdir

### PR DESCRIPTION
**Pull request by pavanvidem**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by bgruening** [comment]
@pavanvidem your medaka commit is still in here. Can you please clean that.

ping @natefoo

File: tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml [hunk 1]
```diff
@@ -12,16 +12,15 @@
     @PREPARE_IDX_MULTIPLE@
 
     #if $reference_source.reference_source_selector == "history":
-        ln -s '${reference_source.ref_file}' &&
-        samtools faidx `basename '${reference_source.ref_file}'` &&
+        #set ref_fa = 'ref.fa'
+        ln -s '${reference_source.ref_file}' 'ref.fa' &&
+        samtools faidx ref.fa &&
+    #else:
+        #set ref_fa = str( $reference_source.ref_file.fields.path )
```
Comment (bgruening) [inline tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml:19]: is the str() needed here?

Comment (wm75) [reply tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml:19]: also, `ref.fa` doesn't need quoting, but `$reference_source.ref_file.fields.path` might.

Comment (pavanvidem) [reply tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml:19]: maybe. I just copied it from calmd here https://github.com/galaxyproject/tools-iuc/blob/11249b49794feb4e39965f09d5420ae3a3dfe2a7/tool_collections/samtools/samtools_calmd/samtools_calmd.xml#L16

**Comment by bernt-matthias** [comment]
Maybe we should integrate this in https://github.com/galaxyproject/tools-iuc/pull/3740 .. and check if other samtools wrappers do the same mistake.

**Comment by bernt-matthias** [comment]
Ideally the tool should use the `@PREPARE_FASTA_IDX@` token? Probably the conditional and maybe in included parameters need to be renamed.

`sam_to_bam` and `calmd` could be treated the same way.

**Comment by bernt-matthias** [comment]
I will merge this into https://github.com/galaxyproject/tools-iuc/pull/3740

**Comment by wm75** [comment]
@bernt-matthias I had assumed the missing version bump was on purpose here to fix the tool *silently*?
Merging things into the larger samtools update will leave public servers with an executable tool version that misbehaves.

@bgruening @natefoo am I misunderstanding things?

**Comment by bernt-matthias** [comment]
@wm75 .. I see. Then lets merge, or @bgruening?

**Comment by bernt-matthias** [comment]
The changes that I have suggested are now implemented in https://github.com/galaxyproject/tools-iuc/pull/3740 

I also cherry-picked the commit from the PR which should not hurt .. maybe I need to merge after this one is in.

**Comment by bgruening** [comment]
ping @pavanvidem

**Comment by pavanvidem** [comment]
I also implemented the suggested changes today morning. I'm pushing my changes now, then I will check if there something that I can adapt from https://github.com/galaxyproject/tools-iuc/pull/3740

**Comment by bernt-matthias** [comment]
> I also implemented the suggested changes today morning. I'm pushing my changes now, then I will check if there something that I can adapt from #3740

I would suggest to keep it as is, i.e. https://github.com/pavanvidem/tools-iuc/commit/b12876b4465fbb6803688fb6f45bab0df8ae357a 

Otherwise we might need a version bump and loose the possibility for a "silent update".

**Comment by pavanvidem** [comment]
The changes shouldn't affect any functionality. Can't we do a silent update here?

**Comment by bernt-matthias** [comment]
> The changes shouldn't affect any functionality. Can't we do a silent update here?

Renaming parameters might affect workflows. So I would say: No

**Comment by pavanvidem** [comment]
Agree, makes sense.

**Review by bernt-matthias — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Good to go @bgruening and @wm75?

**Review by wm75 — APPROVED** [review]
LGTM. Sorry for the back and forth here @pavanvidem

**Comment by pavanvidem** [comment]
all good @wm75, thank you and @bernt-matthias for the review

**Comment by bernt-matthias** [comment]
I will merge this if @bgruening does not object..

**Pull request merged**